src/hints/mod.rs

```diff
@@ -259,9 +277,87 @@ pub fn load_next_tx(
     ap_tracking: &ApTracking,
     _constants: &HashMap<String, Felt252>,
 ) -> Result<(), HintError> {
-    let mut transactions = exec_scopes.get::<Iter<InternalTransaction>>("transactions")?;
+    let mut transactions = exec_scopes.get::<IntoIter<InternalTransaction>>("transactions")?;
     // Safe to unwrap because the remaining number of txs is checked in the cairo code.
     let tx = transactions.next().unwrap();
     exec_scopes.insert_value("transactions", transactions);
+    exec_scopes.insert_value("tx", tx.clone());
     insert_value_from_var_name("tx_type", Felt252::from_bytes_be(tx.r#type.as_bytes()), vm, ids_data, ap_tracking)
 }
+
+/// Implements hint:
+///
+/// ids.contract_address_salt = tx.contract_address_salt
+/// ids.class_hash = tx.class_hash
+/// ids.constructor_calldata_size = len(tx.constructor_calldata)
+/// ids.constructor_calldata = segments.gen_arg(arg=tx.constructor_calldata)
+pub fn prepare_constructor_execution(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let tx = exec_scopes.get::<InternalTransaction>("tx")?;
+    // Safe to unwrap because the remaining number of txs is checked in the cairo code.
+    insert_value_from_var_name(
+        "contract_address_salt",
+        tx.contract_address_salt.unwrap_or(Felt252::from(0)),
+        vm,
+        ids_data,
+        ap_tracking,
+    )?;
+    insert_value_from_var_name("class_hash", tx.class_hash.unwrap_or(Felt252::from(0)), vm, ids_data, ap_tracking)?;
```

Comment:
same here

Reply:
This one fails if I don't return 0 - so class_hash is needed but is not available. The above information would help here as well

Reply:
this still need to be changed to `contract_hash` to match the input JSON